packages/react/src/lib/commonPropInterfaces.ts

```diff
@@ -29,32 +29,7 @@ export interface UIComponentProps<P = any, V = any>
 
 export type SizeValue = 'smallest' | 'smaller' | 'small' | 'medium' | 'large' | 'larger' | 'largest'
 
-export type ColorValue =
-  | 'primary'
-  | 'secondary'
-  | 'blue'
-  | 'green'
-  | 'grey'
-  | 'orange'
-  | 'pink'
-  | 'purple'
-  | 'teal'
-  | 'red'
-  | 'yellow'
-  | string
-
-export type ColorValuesWithPrimitiveColors = ColorValue | 'black' | 'white'
-
-export type ComplexColorPropType<TColorValue = ColorValue> =
-  | {
-      foreground?: TColorValue
-      background?: TColorValue
-      border?: TColorValue
-      shadow?: TColorValue
-    }
-  | TColorValue
-
-export interface ColorComponentProps<TColor = ColorValue> {
+export interface ColorComponentProps<TColor = string> {
```

Comment:
The color prop is simple string - it is theme's responsibility of how it will be handle.